lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,522 @@
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import { render } from '@testing-library/react';
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
+
+import userEvent from '@testing-library/user-event';
+
+import NxCombobox, { Props } from '../NxCombobox';
+
+describe('NxCombobox', function() {
+  const minimalProps: Props<string | number> = {
+        value: '',
+        onChange: () => {},
+        onSearch: () => {},
+        matches: []
+      },
+      quickRender = rtlRender(NxCombobox, minimalProps),
+      renderEl = rtlRenderElement(NxCombobox, minimalProps);
+
+  it('sets specified classNames and attributes on the top-level element', function() {
+    const component = renderEl({ className: 'foo', title: 'bar' });
+
+    expect(component).toHaveClass('foo');
+    expect(component).toHaveAttribute('title', 'bar');
+  });
+
+  it('adds the ref to the top-level element', function() {
+    const ref = React.createRef<HTMLDivElement>(),
+        domNode = render(<NxCombobox { ...minimalProps } ref={ref} />).container.firstElementChild!;
+
+    expect(ref.current).toBeDefined();
+    expect(ref.current).toBe(domNode);
+  });
+
+  it('renders an input child with the combobox role', function() {
+    const inputElement = quickRender().getByRole('combobox');
+
+    expect(inputElement.tagName).toBe('INPUT');
+    expect(inputElement).toBeInTheDocument();
+  });
+
+  it('sets the value as the value of the input', function() {
+    expect(quickRender({ value: 'foo' }).getByRole('combobox')).toHaveValue('foo');
+  });
+
+  it('sets aria-autocomplete on the input to list as default and to both if `autoComplete` prop is set to true',
+      function() {
+        const { getByRole, rerender } = quickRender(),
+            inputElement = getByRole('combobox');
+
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'list');
+
+        rerender(<NxCombobox { ...minimalProps } autoComplete={true}/>);
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'both');
+      });
+
+  it('sets a completion string of the selected suggestion from matches when `autoComplete` prop is set to true',
+      async function() {
+        const { getByRole, rerender } = quickRender({
+              autoComplete: true,
+              matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }]
+            }),
+            inputElement = getByRole('combobox');
+
+        await userEvent.type(inputElement, 'f');
+        rerender(
+          <NxCombobox { ...minimalProps }
+                      autoComplete={true}
+                      matches={ [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }] }/>);
+        expect(inputElement).toHaveValue('Foo');
```

Comment:
May want to use `window.getSelection()` to check which part is the autocomplete

Reply:
> May want to use window.getSelection() to check which part is the autocomplete

Actually it sounds like that likely won't work: https://developer.mozilla.org/en-US/docs/Web/API/Window/getSelection#related_objects